### PR TITLE
Better initialisable proxy

### PR DESCRIPTION
**Pull request by Amxx**
integration of erc1836 like proxy for a lighter deployment and better upgradability. OPS & Relayer updated to deploy, tested with a custom frontend.

Tests are not written yet, will be doing that soon. In the meantime you can have a look at the logic.

### CODE REVIEW

**Comment by Amxx** [comment]
Most tests are done. Next is:
→ review changes
→ update universal-login-example
→ what is LegacyENS for? is it just for testing? Right now it doesn't have a master contract and thus doesn't work with proxies. (hence the initialisation difference in the relayer's WalletService)

**Comment by Amxx** [comment]
@marekkirejczyk

**Comment by marekkirejczyk** [comment]
Hi, looks promising. 
I'll be doing Review today/tomorrow.
Would be good to fix linter.

File: universal-login-contracts/contracts/ENSRegistered.sol [hunk 1]
```diff
@@ -9,7 +9,7 @@ import "./ENS/ReverseRegistrar.sol";
 contract ENSRegistered {
     bytes32 constant ADDR_REVERSE_NODE = 0x91d1777781884d03a6757a803996e38de2a42967fb37eeaca72729271025a9e2;
 
-    constructor(bytes32 _hashLabel, string memory _name, bytes32 _node, ENS ens, FIFSRegistrar registrar, PublicResolver resolver) public {
+    function ENSregister(bytes32 _hashLabel, string memory _name, bytes32 _node, ENS ens, FIFSRegistrar registrar, PublicResolver resolver) internal {
```
Comment (marekkirejczyk) [inline universal-login-contracts/contracts/ENSRegistered.sol]: `registerENS`

File: universal-login-contracts/contracts/common/IMaster.sol [hunk 2]
```diff
@@ -0,0 +1,13 @@
+pragma solidity ^0.5.0;
+
+
+interface IMaster {
+    function owner       ()                            external view returns (address);
+    function master      ()                            external view returns (address);
+    function masterId    ()                            external pure returns (bytes32);
```
Comment (marekkirejczyk) [inline universal-login-contracts/contracts/common/IMaster.sol]: I would propose to remove ERC725 v2 related functions from IMaster and default Wallet.

**Comment by marekkirejczyk** [comment]
Continued in #389

**Pull request closed without merging**